src/components/Admin/Approval/Approval.js

```diff
@@ -4,68 +4,68 @@ import PropTypes from 'prop-types';
 import ItemSelector from '../ItemSelector';
 import DataController from '../utils/DataController';
 
-import './Approval.css';
+const IMAGE_QUEUE_SIZE = 20;
 
 class ImageApproval extends Component {
   state = {
     loading: true,
+    image: null,
     images: [],
-    storeList: [{name: 'unknown', id: 'unknown'}]
+    storeList: {unknown: {name: 'Unknown', id: 'unknown'}}
   };
 
   getImages = async () => {
     return this.dataController
-      .getImages(false, 10)
+      .getImages(false, IMAGE_QUEUE_SIZE)
       .then(images => this.setState({images}));
   };
 
   lastImageTimestamp = () => {
-    return this.state.images.reduce(
-      (highest, current) => Math.max(highest, current.timestamp),
-      0
-    );
+    return Math.max(...this.state.images.map(i => i.timestamp));
   };
 
-  nextImage = index => {
-    return this.dataController
-      .getImages(false, 1, this.lastImageTimestamp())
-      .then(images => {
-        this.setState(prevState => {
-          if (images.length < 1) {
-            prevState.images.splice(index, 1);
-          } else {
-            const [image] = images;
-            if (prevState.images.filter(i => i.id === image.id).length > 0) {
-              this.getImages();
-            } else {
-              prevState.images[index] = image;
-            }
-          }
-          return prevState;
+  displayNextImage = () => {
+    this.setState(prevState => {
+      return {
+        image: prevState.images.shift() || null,
+        images: prevState.images
```

Comment:
There's a bunch of mutation going on here which could cause unintended side effects/make this a bit harder to follow.

Looks like the old code was doing it, but it'd be good to avoid doing this.

Could avoid using `shift` by doing something like...

```js
this.setState(prevState => {
  const [image, ...images] = prevState.images

  return {
    image: image || null,
    images,
  }
})
```